org.jacoco.core.test/src/org/jacoco/core/internal/analysis/ClassAnalyzerTest.java

```diff
@@ -87,4 +87,38 @@ public void testMethodFilter_Lambda() {
 		assertEquals(1, coverage.getMethods().size());
 	}
 
+	@Test
+	public void testMethodFilter_EnumValues() {
+		analyzer.visit(Opcodes.V1_5, Opcodes.ACC_PUBLIC, "Foo", null,
+				"java/lang/Enum", null);
+
+		MethodProbesVisitor mv = analyzer.visitMethod(0, "values", "()[LFoo;",
```

Comment:
I hesitated to create separate, because then IMO not so easy to catch difference between them. But ok, I'll split.